### PR TITLE
fix: I find out a smart way to fix scale problem

### PR DESCRIPTION
**Pull request by Decodetalkers**

### CODE REVIEW

**Comment by Decodetalkers** [comment]
I find that the color on my wm is a little strange. on layershell, since the color is Xrgb8888, maybe we should adjust the data in buffer first then do screenshot?

**Comment by AndreasBackx** [comment]
@Shinyzenith would be better for review since I'm lacking in a bit of time. Though this _seems_ fine from first glance. The colour issue should likely fixed in another diff?

File: libwayshot/src/lib.rs [hunk 1]
```diff
@@ -832,6 +835,12 @@ impl WayshotConnection {
                 ));
             }
         };
+        let viewporter = self.globals.bind::<WpViewporter, _, _>(&qh, 1..=1, ()).ok();
+        if viewporter.is_none() {
+            tracing::info!(
+                "Your wm does not support wp_viewporter, so the scale may come out with some problems"
```
Comment (Shinyzenith) [inline libwayshot/src/lib.rs:841]:
```suggestion
                "Compositor does not support wp_viewporter, display scaling may be inaccurate."
```

**Review by Shinyzenith — APPROVED** [review]

File: libwayshot/src/lib.rs (hunk 1 above)
Comment (Shinyzenith) [reply libwayshot/src/lib.rs:841]: Apart from just the error message, everything else looks good. We can merge

**Comment by Decodetalkers** [comment]
Ok, I will merge it, ☺️

**Pull request merged**